libs/designer-ui/src/lib/monitoring/retrypanel/index.tsx

```diff
@@ -20,6 +20,11 @@ export const RetryPanel: React.FC<RetryPanelProps> = ({ retryHistories, visible
     return calculateDuration(retryHistory.startTime, retryHistory.endTime);
   }, [retryHistory.endTime, retryHistory.startTime]);
 
+  useEffect(() => {
+    const newRetryHistory = retryHistories[currentPage - 1];
```

Comment:
Would there be a reason why we are not using useMemo for this and using useEffect to update the value here?

Reply:
I agree, that would probably be better and lead to fewer rerenders